main.go

```diff
@@ -60,6 +60,7 @@ type genOptions struct {
 	NoEmbedIDL        bool   `long:"no-embed-idl" description:"Do not embed IDLs into the generated code."`
 	NoZap             bool   `long:"no-zap" description:"Do not generate code for Zap logging."`
 	OutputFile        string `long:"output-file" value-name:"FILENAME" description:"Generates a single .go file as an output. Specifying an OutputFile prevents code generation for included Thrift Files."`
+	ErrOnUnknownEnum  bool   `long:"error-on-unknown-enum" value-name:"ERRUNKNOWNENUM" description:"Send error on unknown enum marshal"`
```

Comment:
I'm reluctant to advertise this flag with this description. Marshaling is part of Thrift's problem statement, and this flag has no effect on the *Thrift* marshaling. It only affects the unsupported JSON/text marshaling of enums.

How about:

- make the flag hidden (`hidden:"true"` like the GeneratePluginAPI flag above) so that only people like yourself who know about it can use it
- include "text-marshal" somewhere in the name (`--enum-text-marshal-strict`?)

Also, the `value-name:"..."` option isn't needed for a boolean flag—it specifies the name for the argument of the flag, if any, and booleans don't usually take arguments.